gemma.cc

```diff
@@ -633,30 +633,32 @@ void ForEachTensor(const Weights<TConfig>* weights,
        c_weights.c_final_norm_scale);
 
   char name[16];
-  for (size_t layer_idx = 0; layer_idx < TConfig::kLayers; ++layer_idx) {
-    Layer<TConfig>* layer = weights ? &weights->layers[layer_idx] : nullptr;
-    CompressedLayer<TConfig>* c_layer = c_weights.CLayer(layer_idx);
+  for (int layer_idx = 0; layer_idx < static_cast<int>(TConfig::kLayers);
```

Comment:
what's the motivation here for having int layer_idx and a size_t idx vs the previous size_t layer_idx?

Reply:
snprintf requires an int, but the two functions below want size_t. Making the loop counter size_t would require many casts, one per snprintf.